FirebaseStorageSwift/Sources/StorageComponent.swift

```diff
@@ -0,0 +1,80 @@
+// Copyright 2022 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+import Foundation
+import FirebaseAppCheckInterop
+import FirebaseAuthInterop
+import FirebaseCore
+import FirebaseCoreInternal
+
+@objc(FIRStorageProvider)
+protocol StorageProvider {
+  @objc func Storage(for bucket: String) -> Storage
+  // TODO: See if we can avoid the `type` parameter by either making it a `Storage` argument to
+  // allow subclasses, or avoid it entirely and fix tests. This was done for StorageCombineUnit,
+  // although we may be able to now port to using `@testable` instead of using the mock.
```

Comment:
I'm not sure I 100% follow the TODO but could something like this help?:
```swift 
@objc
protocol StorageProtocol {}

@objc class Storage: NSObject, StorageProtocol {
}

@objc
protocol StorageProvider {
  @objc func storage(for bucket: String) -> StorageProtocol
}
```

`StorageProtocol` could allow for easier fakes.

Reply:
cc: @ryanwilson